### ISSUE TITLE
Fastai LR finder 

### ISSUE THREAD

**Issue by saruarlive** [pytorch/ignite#1482]
Hi, @vfdev-5 et al,
The FastaiLR finder always provides a warning message as shown below,
![image](https://user-images.githubusercontent.com/24542592/100640874-2c09f500-3337-11eb-9879-ce8f75f9e948.png)

What is the issue here? Why is that so? I am looking forward to hearing from you.

**Comment by vfdev-5** [comment]
Hi @saruarlive, yes, current implementation of `lr_finder` is not that user friendly and may not work as expected. Looking at your logs, seems like the batch loss is already `nan` but the warning says that trainer could not detect that.
Try to set `num_iter` to a smaller value, e.g. `len(train_loader)` and use `max_epochs=1` as probably there is no point to run for 50 epochs to find out the best learning rate. Maybe, try also to decrease `end_lr`, `end_lr=5.0` or `end_lr=1.0`.

**Comment by saruarlive** [comment]
Dear @vfdev-5 et al, thanks for your reply.
I have set the parameter as per your suggestions, still, a few iterations at the end provide nan values. I am not sure whether that is okay or not for the LRFinder class.

![image](https://user-images.githubusercontent.com/24542592/100664331-bd895f00-3357-11eb-843d-4811dadf750b.png)

**Comment by vfdev-5** [comment]
Hi @saruarlive , seems like your network is sensible to "large" values of the learning rate.
According to the plot, the loss decreases rapidly when lr is around 10^-2.
Please, try to progressively reduce more `end_lr`, e.g. `end_lr=1.0`, `end_lr=0.5`, `end_lr=0.1`.
Maybe, you can also try to vary `step_mode` as "linear" instead of "exp"...
Also, increasing `num_iter`(e.g. `2 * len(train_loader)`) can better sample learning rate values before it goes nan...

**Comment by vfdev-5** [comment]
@saruarlive I close this issue as solved, please feel free to reopen if need more support on this topic.
